File: 2013/Practice Round/Captain_Hammer.py
# ---------- I/O setup ----------
import sys, os

# get input file paths
current_directory = os.path.dirname(os.path.realpath(__file__))
input_path = os.path.join(current_directory, "Hammer_input.txt")

# redirect stdin file handlers to txt file
sys.stdin=open(input_path, 'r')

# ---------- Solution ----------

# first use time = distance / horizontal speed
# then use SUVAT eqn for vertical motion v = u + at
# eliminate t and rearrange for theta (use sine double angle formula)

import math

# get number of test cases
T = int(input())

for t in range(1, T+1):
    # get launch speed and distance travelled for each test
    V, D = [int(i) for i in input().split()]

    # calculate theta (degrees between 0 and 90)
    theta = 180/math.pi * abs(0.5 * math.asin(-9.8*D / (V**2)))

    # return to 7 decimal places
    theta = format(theta, '.7f')

    # print desired output
    print("Case #{}: {}".format(t, theta))

File: 2013/Practice Round/CPP/Captain_Hammer_cpp/Captain_Hammer_cpp/Captain_Hammer.cpp
#define _USE_MATH_DEFINES
#include <iostream>     // includes cin to read from stdin and cout to write to stdout
#include <cmath>
#include <iomanip>      // std::setprecision
using namespace std;    // since cin and cout are both in namespace std, this saves some text

int main() {
    int T, V, D;
    double theta;

    // get number of test cases
    cin >> T;

    for (int i = 1; i <= T; ++i) {
        // get launch speed and distance travelled for each test
        cin >> V >> D;

        // calculate theta(degrees between 0 and 90)
        theta = 180 / M_PI * abs(0.5 * asin(-9.8 * D / (pow(V, 2))));

        cout << fixed << setprecision(7);
        cout << "Case #" << i << ": " << theta << endl;
    }
    return 0;
}